### ISSUE TITLE
Child died unexpectedly. Packets may have not been sent

### ISSUE THREAD

**Issue by rafajot** [secdev/scapy#780]
root@SG1TO0112301:/tmp# scapy
Welcome to Scapy (2.3.3.dev579)
\>>> ans, unas = sr(IP(dst="110.228.96.179")/ICMP(),timeout=0.8)
Begin emission:
..WARNING: Child died unexpectedly. Packets may have not been sent 22056
Finished to send 1 packets.
*WARNING: Child died unexpectedly. Packets may have not been sent 22047

This kind of error I'm getting with current version 2.3.3.dev579 and its since the commit 4ac969c8a89a50c72f91347b06a1cfffee1c9829 .
I don't have such issue with 1186356a1d73fd59b700d8af05e789d0e8899de6 and before.

**Comment by p-l-** [comment]
Hi,

Can you report your exact configuration (OS & version, arch, conf.L{2,3}socket, Python version)?

**Comment by rafajot** [comment]
I got this error on every Linux I tried so I (probably too optimistically) assumed it's not very environment specific issue.
Linux 3.13.0-65-generic #106-Ubuntu SMP Fri Oct 2 22:08:27 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
Python 2.7.6

I'm not quite sure how to obtain conf.L{2,3}socket details. Is that it:
>>> conf.L3socket
<L3PacketSocket: read/write packets at layer 3 using Linux PF_PACKET sockets>
>>> print conf.L3socket
<L3PacketSocket: read/write packets at layer 3 using Linux PF_PACKET sockets>
?

**Comment by gpotter2** [comment]
Cannot reproduce on Windows...

**Comment by p-l-** [comment]
Can you set the environment variable `SCAPY_USE_PCAPDNET=yes` and try again?:
```
# SCAPY_USE_PCAPDNET=yes scapy
Welcome to Scapy (2.3.3.dev579)
>>>
```

**Comment by rafajot** [comment]
Still doesn't look good:

root@SG1TO0112301:~# SCAPY_USE_PCAPDNET=yes scapy
ERROR: Unable to import pcap module: No module named pcap/No module named pcapy
ERROR: Unable to import dnet module: No module named dumbnet
WARNING: No route found for IPv6 destination :: (no default route?)
Welcome to Scapy (2.3.3.dev579)
\>>> ans, unas = sr(IP(dst="8.8.8.8")/ICMP(),timeout=0.8)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'IP' is not defined
\>>>

So is that only me having this issue on Linux?

**Comment by p-l-** [comment]
You need to install the Python packages pcap or pcapy, and dnet I guess. I don't know if that's **only** you, but I cannot reproduce this issue on my linux computer.

**Comment by p-l-** [comment]
Can you also try to apply the patch from #770 and try again (without the environment variable set)?

**Comment by rafajot** [comment]
Now (after applying patch) it looks good:

root@SG1TO0112301:/tmp# scapy
Welcome to Scapy (2.3.3.dev579)
\>>> ans, unas = sr(IP(dst="8.8.8.8")/ICMP(),timeout=0.8)
Begin emission:
..Finished to send 1 packets.
*
Received 3 packets, got 1 answers, remaining 0 packets
\>>>

**Comment by p-l-** [comment]
I don't know what the problem was, but since it seems fixed by #770 we are not going to investigate this issue. It will be closed when the patch is merged.
